builder/alicloud/ecs/step_config_alicloud_security_group.go

```diff
@@ -0,0 +1,136 @@
+	client := state.Get("client").(*ecs.Client)
+	ui := state.Get("ui").(packer.Ui)
+	networkType := state.Get("networktype").(InstanceNetWork)
+
+	var securityGroupItems []ecs.SecurityGroupItemType
+	var err error
+	if len(s.SecurityGroupId) != 0 {
+		if networkType == VpcNet {
+			vpcId := state.Get("vpcid").(string)
+			securityGroupItems, _, err = client.DescribeSecurityGroups(&ecs.DescribeSecurityGroupsArgs{
+				VpcId:    vpcId,
+				RegionId: common.Region(s.RegionId),
+			})
+		} else {
+			securityGroupItems, _, err = client.DescribeSecurityGroups(&ecs.DescribeSecurityGroupsArgs{
+				RegionId: common.Region(s.RegionId),
+			})
+		}
+
+		if err != nil {
+			ui.Say(fmt.Sprintf("Query alicloud security grouip failed: %s", err))
+			state.Put("error", err)
+			return multistep.ActionHalt
+		}
+		for _, securityGroupItem := range securityGroupItems {
+			if securityGroupItem.SecurityGroupId == s.SecurityGroupId {
+				state.Put("securitygroupid", s.SecurityGroupId)
+				s.isCreate = false
+				return multistep.ActionContinue
+			}
+		}
+		s.isCreate = false
+		message := fmt.Sprintf("The specific security group {%s} isn't exist.", s.SecurityGroupId)
+		state.Put("error", errors.New(message))
+		ui.Say(message)
+		return multistep.ActionHalt
+
+	}
+	var securityGroupId string
+	ui.Say("Start creating security groups...")
+	if networkType == VpcNet {
+		vpcId := state.Get("vpcid").(string)
+		securityGroupId, err = client.CreateSecurityGroup(&ecs.CreateSecurityGroupArgs{
+			RegionId:          common.Region(s.RegionId),
+			SecurityGroupName: s.SecurityGroupName,
+			VpcId:             vpcId,
+		})
+	} else {
+		securityGroupId, err = client.CreateSecurityGroup(&ecs.CreateSecurityGroupArgs{
+			RegionId:          common.Region(s.RegionId),
+			SecurityGroupName: s.SecurityGroupName,
+		})
+	}
+	if err != nil {
+		state.Put("error", err)
+		ui.Say(fmt.Sprintf("Create security group failed %v", err))
+		return multistep.ActionHalt
+	}
+	state.Put("securitygroupid", securityGroupId)
+	s.isCreate = true
+	s.SecurityGroupId = securityGroupId
+	err = client.AuthorizeSecurityGroupEgress(&ecs.AuthorizeSecurityGroupEgressArgs{
+		SecurityGroupId: securityGroupId,
+		RegionId:        common.Region(s.RegionId),
+		IpProtocol:      ecs.IpProtocolAll,
+		PortRange:       "-1/-1",
+		NicType:         ecs.NicTypeInternet,
+		DestCidrIp:      "0.0.0.0/0", //The input parameter "DestGroupId" or "DestCidrIp" cannot be both blank.
+	})
+	if err != nil {
+		state.Put("error", err)
+		ui.Say(fmt.Sprintf("authorzie security group failed %v", err))
+		return multistep.ActionHalt
+	}
+	err = client.AuthorizeSecurityGroup(&ecs.AuthorizeSecurityGroupArgs{
+		SecurityGroupId: securityGroupId,
+		RegionId:        common.Region(s.RegionId),
+		IpProtocol:      ecs.IpProtocolAll,
+		PortRange:       "-1/-1",
+		NicType:         ecs.NicTypeInternet,
+		SourceCidrIp:    "0.0.0.0/0", //The input parameter "SourceGroupId" or "SourceCidrIp" cannot be both blank.
+	})
```

Comment:
You are right, but most of our user isn't familiar with how to configure the security group, if we only allow limited traffics, it's most probability fail for the builder. For those know how to configure security group, they can add rule themselves.  At the sametime, the build process will last a short time and after the build, it will be destroyed. I think after user become familiar with it, we can change the default rule only allow the requred traffics. What's your though?